File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys

N = int(sys.stdin.readline())
a = sorted(list(map(int, sys.stdin.readline().split())))
x = int(sys.stdin.readline())

result = 0
left, right = 0, N-1

while left < right:
  x1 = a[left] + a[right]
  if x1 == x:
    result += 1
    left += 1
  elif x1 < x:
    left += 1
  else:
    right -= 1

print(result)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.swift
let N = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map { Int($0)! }
let X = Int(readLine()!)!

let sortedArr = arr.sorted()

var left = 0
var right = N - 1

var currentSum = 0
var result = 0

while left < right {
    currentSum = sortedArr[left] + sortedArr[right]
    if currentSum == X {
        result += 1
        left += 1
    } else if currentSum < X {
        left += 1
    } else {
        right -= 1
    }
}

print(result)
